common/fs/src/cache/tailed_file.rs

```diff
@@ -279,7 +279,11 @@ impl IngestLineSerialize<String, bytes::Bytes, std::collections::HashMap<String,
     where
         S: SerializeUtf8<bytes::Bytes> + std::marker::Send,
     {
-        let bytes = {
+        // Try to use the cached value first
+        let replaced_buf = self.line_buffer.take();
```

Comment:
If self.line_buffer is a Bytes object then we could replace this take with a clone() as Bytes has a very cheap Clone impl